### ISSUE TITLE
Ubuntu install has unmet dependencies[?]

### ISSUE THREAD

**Issue by SvenGeier** [3Dickulus/FragM#129]
**Describe the bug**
Downloaded the .deb from this site:

Fragmentarium-2.5.3-200315-Linux.deb

When I try to install it, this happens:

`% sudo apt install -f ./Fragmentarium-2.5.3-200315-Linux.deb
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'fragmentarium' instead of './Fragmentarium-2.5.3-200315-Linux.deb'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 fragmentarium : Depends: qtbase5 but it is not installable
                 Depends: qtscript5 but it is not installable
                 Depends: qtxml5 but it is not installable
E: Unable to correct problems, you have held broken packages.`

Indeed, the package search tool does not find these named packages:
[qtscript5](https://packages.ubuntu.com/search?keywords=qtscript5)
[qtbase5](https://packages.ubuntu.com/search?keywords=qtbase5)
etc

This is on Mint:
`% uname -a`
`Linux Penguin 5.3.0-45-generic #37~18.04.1-Ubuntu SMP Fri Mar 27 15:58:10 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by 3Dickulus** [comment]
The build system...
```
Build language: cpp
Build group: stable
Build dist: xenial
Build id: 662810514
Job id: 662810515
Runtime kernel version: 4.15.0-1028-gcp
travis-build version: 681093eb9
Build image provisioning date and time
Mon Mar 25 16:43:24 UTC 2019
Operating System Details
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04
Codename:	xenial
```

Installed software...

```
    sudo apt-get -qq install cmake
    sudo apt-get -qq install extra-cmake-modules
    sudo apt-get -qq install libopenexr-dev
    sudo apt-get -qq install libilmbase-dev
    sudo apt-get -qq install libqt5scripttools5
    sudo apt-get -qq install libqt5gui5
    sudo apt-get -qq install libqt5libqgtk2
    sudo apt-get -qq install libqt5svg5-dev
    sudo apt-get -qq install libqt5opengl5-dev
    sudo apt-get -qq install libqt5xmlpatterns5-dev
    sudo apt-get -qq install qt5-qmake
    sudo apt-get -qq install qt5-default
    sudo apt-get -qq install qttools5-dev-tools
    sudo apt-get -qq install qtbase5-dev
    sudo apt-get -qq install qtbase5-private-dev
    sudo apt-get -qq install qtbase5-dev-tools
    sudo apt-get -qq install qt5-style-plugins
    sudo apt-get -qq install qt5-image-formats-plugins
    sudo apt-get -qq install qtscript5-dev
    
    wget https://launchpad.net/ubuntu/+archive/primary/+files/libglm-dev_0.9.9.3-2_all.deb
    sudo dpkg -i libglm-dev_0.9.9.3-2_all.deb
```

Sorry I don't know anything about Mint ??? all I can tell you is what's on the Travis-CI machine that builds the package :-\

**Comment by 3Dickulus** [comment]
just doing some testing
should have "working" release packages ready next day or 2.

**Comment by 3Dickulus** [comment]
just generated fresh packages

debian depends: libopenexr24, libqt5opengl5, libqt5gui5, libqt5script5, libqt5xml5

**Comment by retrue** [comment]
The deb package available here in resources doesn't work with the more recent Ubuntu or similar distros.
I tried to compile and install from Linux Mint 18.3.
I couldn't install the ppa because it is for Xenial or older. I couldn't install libopenerxr24 for the same reason.
Anyway, I installed all the packages in the instructions and I got two more mistakes for missing dependencies.
- An  error related to Qt5LinguistTools that was solved by installing:
sudo apt install libqt5svg5-dev
sudo apt-get install qttools5-dev
- An error related to / usr / bin / ld, which was solved installing:
sudo apt-get install zlib1g-dev 

After that I was able to complete the installation process without error messages, still the installation is a failure. Not all the necessary files are copied to the folder Fragmentarium-2.5.3 in my home and the Fragmentarium-2.5.3 executable doesn't work.
So, maybe missing dependencies, wrong versions or whatever.
Some detailed instructions to make an installation possible in Ubuntu 18.04? or the next 20.04 that will be out very soon?
Thanks.

**Comment by 3Dickulus** [comment]
guessing those get pulled in as dependencies when other qt-dev packages are installed on the Travis-CI server

have you tried the deb package on the OS it was compiled for?

Operating System Details
Distributor ID: Ubuntu
Description: Ubuntu 16.04.6 LTS
Release: 16.04
Codename: xenial

for any other OS or variant I expect that it would fail, will look at using 18.04 on Travis-CI for building the next deb package

when I  examine the posted deb file I see all of the distribution files are included

glm is not version dependant, the latest for your OS should be fine
OpenEXR is not version dependant in the cmakelists.txt file or the sources for the linux build
zlib1g-dev is a dependency of OpenEXR

Qt development environment and all associated tool chain files and dependencies should be completely installed before compiling.

**Comment by retrue** [comment]
No, I haven't tried the provided deb in a Xenial installation. Libopenerxr24 is an official mod of the libopenerxr library whose more recent official version is libopenerxr22. The group who made the unofficial version hasn't updated its version in 4 years.
When I compiled Fragmentarium, it got installed in home/my_name/Fragmentarium-2.5.3
But even if I didn't get any error messages the installation was not correct, there were many missing files and the executable didn't work.
I installed all those packages in the main page, except the ppa.
I am not a programmer and I don't use Linux normally. So, if you have any suggestion, I could try to compile it again.

**Comment by 3Dickulus** [comment]
I don't understand "many missing files" you mean system libraries provided by OS or FragM distribution files ?

**Comment by 3Dickulus** [comment]
```
      sudo add-apt-repository -y ppa:ubuntu-sdk-team/ppa
      sudo apt-get update -qq
      sudo apt-get -qq install cmake
      sudo apt-get -qq install extra-cmake-modules
      sudo apt-get -qq install libopenexr-dev
      sudo apt-get -qq install libilmbase-dev
      sudo apt-get -qq install libqt5scripttools5
      sudo apt-get -qq install libqt5gui5
      sudo apt-get -qq install libqt5libqgtk2
      sudo apt-get -qq install libqt5svg5-dev
      sudo apt-get -qq install libqt5opengl5-dev
      sudo apt-get -qq install libqt5xmlpatterns5-dev
      sudo apt-get -qq install qt5-qmake
      sudo apt-get -qq install qt5-default
      sudo apt-get -qq install qttools5-dev-tools
      sudo apt-get -qq install qtbase5-dev
      sudo apt-get -qq install qtbase5-private-dev
      sudo apt-get -qq install qtbase5-dev-tools
      sudo apt-get -qq install qt5-style-plugins
      sudo apt-get -qq install qt5-image-formats-plugins
      sudo apt-get -qq install qtscript5-dev

```

this is what happens on Travis build server then...
```
mkdir Fragmentarium-Source/build
cd Fragmentarium-Source/build
```
you may have to replace the ppa with the equivalent for Mint

**Comment by retrue** [comment]
I don't know which files are included in a correct, working, installation of fragmentarium, but for example, it created the folder "Examples" but it was empty.
I'll try later again and make a list of the files that are created.
About the ppa, there is not equivalente for Mint 19 (Ubuntu 18.04 Bionic). That ppa by Ubuntu-sdk-team only has versions for old versions of Ubuntu (from 14.04 to 17.04). It has not been updated in 3 years.

**Comment by 3Dickulus** [comment]
yes, doing some testing, will update the install instructions and see what I can do about the zip generated by github

I note in your install log "Can't find Fragmentarium_es.qm !!!" would you be interested in providing the Spanish translation?

**Comment by retrue** [comment]
I even thought about doing the Spanish translation for XaoS, that is incomplete.
It is long / difficult to translate Fragmentarium?

**Comment by 3Dickulus** [comment]
in the source tree the translations folder holds the raw files

here is an empty translation file, it has all of the English just needs Spanish
[Fragmentarium_es.zip](https://github.com/3Dickulus/FragM/files/4434586/Fragmentarium_es.zip)

the filename after unzip should be Fragmentarium_es.ts

you can open this file with the Qt tool Linguist, just type `linguist Fragmentarium_es.ts` the tool is quite intuitive, just play around with it and if you make progress with it let me know, if not then no worries.

**Comment by retrue** [comment]
I'll give it a try.

**Comment by 3Dickulus** [comment]
re:OP the front page now has some info about the submodules and how to make sure you get them and the release page now has a zip archive that has the complete sources.

**Comment by 3Dickulus** [comment]
oops, not closed until Travis is building a more recent working deb package

**Comment by retrue** [comment]
I have a MX Linux 19 distro installed in VirtualBox, so I decided to compile Fragmentarium there.
I think the compilation has been sucessful. Of course, Fragmentarium doesn't run inside VirtualBox.
MX Linux 19 is a very popular distribution based on Debian Stable.
The packages list is a bit different than for Ubuntu Xenial or Bionic (Mint 19).

  sudo apt-get -qq install cmake
  sudo apt-get -qq install extra-cmake-modules
  sudo apt-get -qq install libopenexr-dev
  sudo apt-get -qq install libilmbase-dev
  sudo apt-get -qq install libqt5scripttools5
  sudo apt-get -qq install libqt5gui5
  sudo apt-get -qq install libqt5svg5-dev
  sudo apt-get -qq install libqt5opengl5-dev
  sudo apt-get -qq install libqt5xmlpatterns5-dev
  sudo apt-get -qq install qt5-qmake
  sudo apt-get -qq install qttools5-dev-tools
  sudo apt-get -qq install qtbase5-dev
  sudo apt-get -qq install qtbase5-dev-tools
  sudo apt-get -qq install qt5-style-plugins
  sudo apt-get -qq install qt5-image-formats-plugins
  sudo apt-get -qq install qtscript5-dev

Plus:
 sudo apt-get install qttools5-dev
 sudo apt-get install libglm-dev

zlib1g-dev was already installed. I don't know if it was installed by default or when I installed some software, but I think it should be included in the dependencies list.
ligblm-dev can be installed directly from the MX Linux repository and it is already in its last version.

I include the log of the installation.

[MKLinux19_FragM_Installation.log](https://github.com/3Dickulus/FragM/files/4437657/MKLinux19_FragM_Installation.log)
